plugins/inputs/syslog/rfc5426_test.go

```diff
@@ -0,0 +1,276 @@
+			name: "max",
+			data: []byte(fmt.Sprintf("<%d>%d %s %s %s %s %s - %s", maxP, maxV, maxTS, maxH, maxA, maxPID, maxMID, message7681)),
+			wantBestEffort: &testutil.Metric{
+				Measurement: "syslog",
+				Fields: map[string]interface{}{
+					"version":   maxV,
+					"timestamp": time.Unix(1514764799, 999999000).UTC(),
+					"message":   message7681,
+					"procid":    maxPID,
+					"msgid":     maxMID,
+				},
+				Tags: map[string]string{
+					"severity":         "7",
+					"severity_level":   "debug",
+					"facility":         "23",
+					"facility_message": "local use 7 (local7)",
+					"hostname":         maxH,
+					"appname":          maxA,
+				},
+				Time: defaultTime,
+			},
+			wantStrict: &testutil.Metric{
+				Measurement: "syslog",
+				Fields: map[string]interface{}{
+					"version":   maxV,
+					"timestamp": time.Unix(1514764799, 999999000).UTC(),
+					"message":   message7681,
+					"procid":    maxPID,
+					"msgid":     maxMID,
+				},
+				Tags: map[string]string{
+					"severity":         "7",
+					"severity_level":   "debug",
+					"facility":         "23",
+					"facility_message": "local use 7 (local7)",
+					"hostname":         maxH,
+					"appname":          maxA,
+				},
+				Time: defaultTime,
+			},
+		},
+		{
+			name: "minimal/incomplete",
+			data: []byte("<1>2"),
+			wantBestEffort: &testutil.Metric{
+				Measurement: "syslog",
+				Fields: map[string]interface{}{
+					"version": uint16(2),
+				},
+				Tags: map[string]string{
+					"severity":         "1",
+					"severity_level":   "alert",
+					"facility":         "0",
+					"facility_message": "kernel messages",
+				},
+				Time: defaultTime,
+			},
+			werr: true,
+		},
+	}
+
+	return testCases
+}
+
+func newUDPSyslogReceiver(bestEffort bool) *Syslog {
+	return &Syslog{
+		Protocol: "udp",
+		Address:  address,
+		now: func() time.Time {
+			return defaultTime
+		},
+		BestEffort: bestEffort,
+	}
+}
+
+func testRFC5426(t *testing.T, acc *testutil.Accumulator, bestEffort bool) {
+	for _, tc := range getTestCasesForRFC5426() {
+		t.Run(tc.name, func(t *testing.T) {
+			// Clear
+			acc.ClearMetrics()
+			// Connect
+			conn, err := net.Dial("udp", address)
+			defer conn.Close()
+			require.NotNil(t, conn)
+			require.Nil(t, err)
+			// Write
+			_, e := conn.Write(tc.data)
+			require.Nil(t, e)
+			// Waiting ...
+			if tc.wantStrict == nil && tc.werr || bestEffort && tc.werr {
+				acc.WaitError(1)
+			}
+			if tc.wantBestEffort != nil && bestEffort || tc.wantStrict != nil && !bestEffort {
+				acc.Wait(1) // RFC5426 mandates a syslog message per UDP packet
+			}
+			// Compare
+			var got *testutil.Metric
+			var want *testutil.Metric
+			if len(acc.Metrics) > 0 {
+				got = acc.Metrics[0]
+			}
+			if bestEffort {
+				want = tc.wantBestEffort
+			} else {
+				want = tc.wantStrict
+			}
+			if !cmp.Equal(want, got) {
+				t.Fatalf("Got (+) / Want (-)\n %s", cmp.Diff(want, got))
+			}
+		})
+	}
+}
+
+func TestUDPInBestEffortMode(t *testing.T) {
```

Comment:
The way these tests are structured make it more likely that they will fail due to a previous test case, because you are reusing the unit under test.  I would create all tested code new for each test, unless you are specifically testing multiple connections.

Reply:
Should be better now.